core/src/main/java/com/novoda/espresso/AccessibilityRules.java

```diff
@@ -0,0 +1,67 @@
+package com.novoda.espresso;
+
+import android.accessibilityservice.AccessibilityServiceInfo;
+import android.content.Context;
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.view.accessibility.AccessibilityManager;
+
+import org.junit.rules.TestRule;
+
+import java.util.List;
+
+import static com.novoda.espresso.AccessibilityServiceToggler.Service;
+
+public class AccessibilityRules {
+
+    public static TestRule createTalkBackTestRule() {
+        return new AccessibilityServiceTestRule(createHelper(Service.TALKBACK));
+    }
+
+    public static TestRule createSwitchAccessTestRule() {
+        return new AccessibilityServiceTestRule(createHelper(Service.SWITCH_ACCESS));
+    }
+
+    public static TestRule createSelectToSpeakTestRule() {
+        return new AccessibilityServiceTestRule(createHelper(Service.SELECT_TO_SPEAK));
+    }
+
+    private static AccessibilityServiceTestRule.Helper createHelper(final Service service) {
+        final Context context = InstrumentationRegistry.getTargetContext();
+        final AccessibilityManager a11yManager = (AccessibilityManager) context.getSystemService(Context.ACCESSIBILITY_SERVICE);
+
+        return new AccessibilityServiceTestRule.Helper() {
+            @Override
+            public String serviceName() {
+                return service.name();
+            }
+
+            @Override
+            public void requestEnableService() {
+                requestState(AccessibilityServiceTogglingActivity.ENABLED);
```

Comment:
We cannot use the Toggler directly because this code is running in the test apk which does not have permission to write to system settings.

So we fire an intent to the activity which is part of the app under test, and let that one toggle the service.